vortex-datafusion/src/persistent/execution.rs

```diff
@@ -156,12 +156,21 @@ impl ExecutionPlan for VortexExec {
         target_partitions: usize,
         _config: &ConfigOptions,
     ) -> DFResult<Option<Arc<dyn ExecutionPlan>>> {
-        let file_groups = self.file_scan_config.file_groups.clone();
+        // If there's only one total file in the scan, we can't repartition it
+        if self
+            .file_scan_config
+            .file_groups
+            .iter()
+            .map(|group| group.len())
+            .sum::<usize>()
+            == 1
+        {
+            return Ok(None);
```

Comment:
Is this true? Why shouldn't we repartition one single file if it's very big?

Reply:
I guess we can punt this until we implement splitting files into datafusion partitions

Reply:
yeah, its just a missing feature right now but there's no point of doing all the work to repartition a single file back into a single file.